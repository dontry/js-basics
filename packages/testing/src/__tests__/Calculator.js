import React from "react";
import { render } from "react-testing-library";
import Calculator from "../components/Calculator";
import loadable from "react-loadable";

test("renders before loading", () => {
  const { container, debug } = render(<Calculator />);
  expect(container.firstChild).toMatchSnapshot();
});

test("renders after loading", async () => {
  await loadable.preloadAll();
  const { container, debug } = render(<Calculator />);
  expect(container.firstChild).toMatchSnapshot();
});
