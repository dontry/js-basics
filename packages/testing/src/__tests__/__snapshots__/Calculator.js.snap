// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders after loading 1`] = `
.emotion-0 {
  color: white;
  background: #1c191c;
  line-height: 130px;
  font-size: 6em;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
}

.emotion-0 {
  color: white;
  background: #1c191c;
  line-height: 130px;
  font-size: 6em;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
}

<div
  class="calculator"
>
  <div
    class="emotion-0 emotion-1"
  >
    <div
      class="autoScalingText"
      style="transform: scale(1,1);"
    >
      0
    </div>
  </div>
  <div
    class="calculatorKeypad"
  >
    <div
      class="inputKeys"
    >
      <div
        class="functionKeys"
      >
        <button
          class="calculatorKey keyClear"
        >
          AC
        </button>
        <button
          class="calculatorKey keySign"
        >
          ±
        </button>
        <button
          class="calculatorKey keyPercent"
        >
          %
        </button>
      </div>
      <div
        class="digitKeys"
      >
        <button
          class="calculatorKey key0"
        >
          0
        </button>
        <button
          class="calculatorKey keyDot"
        >
          ●
        </button>
        <button
          class="calculatorKey key1"
        >
          1
        </button>
        <button
          class="calculatorKey key2"
        >
          2
        </button>
        <button
          class="calculatorKey key3"
        >
          3
        </button>
        <button
          class="calculatorKey key4"
        >
          4
        </button>
        <button
          class="calculatorKey key5"
        >
          5
        </button>
        <button
          class="calculatorKey key6"
        >
          6
        </button>
        <button
          class="calculatorKey key7"
        >
          7
        </button>
        <button
          class="calculatorKey key8"
        >
          8
        </button>
        <button
          class="calculatorKey key9"
        >
          9
        </button>
      </div>
    </div>
    <div
      class="operatorKeys"
    >
      <button
        class="calculatorKey keyDivide"
      >
        ÷
      </button>
      <button
        class="calculatorKey keyMultiply"
      >
        ×
      </button>
      <button
        class="calculatorKey keySubtract"
      >
        −
      </button>
      <button
        class="calculatorKey keyAdd"
      >
        +
      </button>
      <button
        class="calculatorKey keyEquals"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`renders before loading 1`] = `
<div
  class="calculator"
>
  <div
    style="height: 120px;"
  >
    Loading display...
  </div>
  <div
    class="calculatorKeypad"
  >
    <div
      class="inputKeys"
    >
      <div
        class="functionKeys"
      >
        <button
          class="calculatorKey keyClear"
        >
          AC
        </button>
        <button
          class="calculatorKey keySign"
        >
          ±
        </button>
        <button
          class="calculatorKey keyPercent"
        >
          %
        </button>
      </div>
      <div
        class="digitKeys"
      >
        <button
          class="calculatorKey key0"
        >
          0
        </button>
        <button
          class="calculatorKey keyDot"
        >
          ●
        </button>
        <button
          class="calculatorKey key1"
        >
          1
        </button>
        <button
          class="calculatorKey key2"
        >
          2
        </button>
        <button
          class="calculatorKey key3"
        >
          3
        </button>
        <button
          class="calculatorKey key4"
        >
          4
        </button>
        <button
          class="calculatorKey key5"
        >
          5
        </button>
        <button
          class="calculatorKey key6"
        >
          6
        </button>
        <button
          class="calculatorKey key7"
        >
          7
        </button>
        <button
          class="calculatorKey key8"
        >
          8
        </button>
        <button
          class="calculatorKey key9"
        >
          9
        </button>
      </div>
    </div>
    <div
      class="operatorKeys"
    >
      <button
        class="calculatorKey keyDivide"
      >
        ÷
      </button>
      <button
        class="calculatorKey keyMultiply"
      >
        ×
      </button>
      <button
        class="calculatorKey keySubtract"
      >
        −
      </button>
      <button
        class="calculatorKey keyAdd"
      >
        +
      </button>
      <button
        class="calculatorKey keyEquals"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
